### ISSUE TITLE
Phylo newick parser on negative branch lengths

### ISSUE THREAD

**Issue by mgrigni** [biopython/biopython#664]
Under python 2.7, the 1.65 Newick tree parser is confused by negative branch lengths.
For example:
```
import Bio.Phylo
from cStringIO import StringIO
print "Bio python version", Bio.__version__
T = Bio.Phylo.read(StringIO("(A:1, B:-2, (C:3, D:4):-2)"), 'newick')
for leaf in T.get_terminals():
    d = T.distance(leaf)
    print "leaf at distance", d, "from root has name", leaf.name
```
Correct output:
```
version 1.58
leaf at distance 1.0 from root has name A
leaf at distance -2.0 from root has name B
leaf at distance 1.0 from root has name C
leaf at distance 2.0 from root has name D
```
Incorrect output:
```
version 1.65
leaf at distance 1.0 from root has name A
leaf at distance 0 from root has name -2
leaf at distance 3.0 from root has name C
leaf at distance 4.0 from root has name D
```
I haven't checked 1.66.

**Comment by peterjc** [comment]
Nice to see a self contained test - that should be a big help, possibly it can even be turned into a functional test for the test suite.

**Comment by peterjc** [comment]
Testing with the current code from GitHub under Python 2.7.10

```
Bio python version 1.66+
leaf at distance 1.0 from root has name A
leaf at distance 0 from root has name -2
leaf at distance 3.0 from root has name C
leaf at distance 4.0 from root has name D
```

**Comment by vishnu93** [comment]
Submitted pr #680 . Added some test cases.

**Comment by peterjc** [comment]
Fixed with merging of #680.
